auth/server/src/main/java/com/seagates3/acl/ACLValidation.java

```diff
@@ -224,7 +224,7 @@ static boolean checkIdExists(String canonicalID, String displayname) {
       return false;
     }
 
-    if (ldapResults.hasMore()) {
+    if (ldapResults != null && ldapResults.hasMore()) {
```

Comment:
What action to take if ldapResults is NULL ?

Should we instead of doing if check, add  **catch(NullPointerException e)** ?

Reply:
No. Because of this check it will return false if ldapresult is NULL and we will get expected behavior
Same is the case with all files.
Also, this change is already in R2 we just patching back in main

Reply:
good to hear that we are taking right action if the value is NULL.

But I still believe its better we use catch block instead of checking for NULL in if block.

Reply:
do you have other concern on using catch ?

Reply:
Also Shalaka, where exactly we are returning false if ldapResults is null ??

I see we go to line 242, if ldapResults is set to null !!! Am I missing something ?

Reply:
Catching NPE is not required here. and its not good practice